### ISSUE TITLE
Error on Ipfire 2.21 Core 122

### ISSUE THREAD

**Issue by Smookydope** [stdevel/check_pakfire#8]
Hi there,

on the new version of Ipfire (2.21 Core 122) comes a error when i run the check_pakfire.py script.

```
[root@router ~]# /usr/lib/nagios/check_pakfire.py -d
OPTIONS: {'mirrors': [], 'pkgs_warn': 5, 'show_perfdata': False, 'exclude_pkgs': False, 'pkgs_crit': 10, 'debug': True, 'list_pkgs': False}
RELEASE: 2.21, UPDATE: 122
Traceback (most recent call last):
  File "/usr/lib/nagios/check_pakfire.py", line 213, in <module>
    sys_pkgs = get_local_pkg_versions()
  File "/usr/lib/nagios/check_pakfire.py", line 103, in get_local_pkg_versions
    if my_name != "core-upgrade":
UnboundLocalError: local variable 'my_name' referenced before assignment
```

is there a chance for a fix ?

Thx

Smooky

**Comment by stdevel** [comment]
Hello @Smookydope,
thanks for sharing. I will check and update on this.

Best wishes,
Christian

**Comment by stdevel** [comment]
Hello @Smookydope,
I was unable to reproduce that error on Core 122 and 123. Could you please paste the output of the following commands:

```
$ ./check_pakfire.py --version
$ python --version
$ which python
```

Just want to ensure that you're using the latest version (*1.0.8*) of the plugin. My setup uses Python 2.7.13. Is your IPFire a fresh installation or an older setup with installed updates?

Best wishes,
Christian.

**Comment by stdevel** [comment]
Hello Marcel!

Nevermind - happy to help! Happy to hear that somebody is using my code. 🎉 
Anyhow, let's keep this issue in english if somebody else (*not speaking german*) is having the same issue.

I tried to reproduce the issue on a new installation in a VM - but again this did not work. Anyhow, the error message is quite clear and I wrote a fix in commit b41865ff2a2c099d906fc3f2cc6639b8fcca88b7.

Can you try this script version on your machine?

**Comment by Smookydope** [comment]
Hey Christian,

you add 
`	my_name=""`
on line 95 .

after i add
```
	my_vers=""
	my_rel=""
```
behind you line so it looks like

```
def get_local_pkg_versions():
	#get local installed package versions
	pkgs={}
	my_name=""
	my_vers=""
	my_rel=""
	for root, dirs, files in os.walk("/opt/pakfire/db/installed", topdown=False):
```

it works :-)

Thank you very much.

Best wishes
Marcel

**Comment by stdevel** [comment]
Hey Marcel,
thanks for the hint - fixed this in commit f8113c344dfa5463e37b4e7dc913cde9b29fd172.
